oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -1144,7 +1175,7 @@ impl RuntimeProxy {
     pub fn wait_on_channels(
         &self,
         channel_read_handles: &[Handle],
-    ) -> Result<Vec<ChannelReadStatus>, OakStatus> {
+    ) -> Result<Vec<ChannelReadStatus>, ErroneousChannelStatus> {
```

Comment:
Alternatively, could return `(Vec<ChannelReadStatus>, Result<(), OakStatus>)` -- not sure it's necessarily better though, but something to consider. The idea is that it always returns the vec of statuses, plus either Ok or an error.

Reply:
This would not allow using the `?` operator, so almost every line in the code needs to be changed. Or is there a workaround? 

I would rather have `Result<ChannelStatusVec, OakStatus>` as the return type. `ChannelStatusVec` (or we could call it something else), could be an enum that distinguishes between two outcomes: `ReadReady` and `HasInvalid`. The problem with this solution is that it does not really **fail** if there is an invalid channel in the list, it returns `Ok`. If this is not a big issue, I prefer this alternative.

Reply:
@tiziano88 WDYT?

Reply:
Perhaps let's go with your current approach of extending the error variant to also include the status vector, it makes it clear what the error condition is, even if it means that callers need to handle the special error type manually instead of propagating it with `?` (the point of the change is that there is more information in an error than just `OakStatus`, so it seems fine to me to require extra handling).